File: Fade/Fade/ViewController.m
//
//  ViewController.m
//  Fade
//
//  Created by ZhangYu on 2017/8/29.
//  Copyright © 2017年 DeveloperZhang. All rights reserved.
//

#import "ViewController.h"
#import "Facade.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    Facade *facade = [Facade new];
    [facade methodA];
    [facade methodB];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Interpreter/Interpreter/ViewController.swift
//
//  ViewController.swift
//  Interpreter
//
//  Created by ZhangYu on 2017/9/10.
//  Copyright © 2017年 DeveloperZhang. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        let context = Context();
        let terminalExpression = TerminalExpression();
        let nonTerminalExpression = NonTerminalExpression();
        terminalExpression.interpret(context: context);
        nonTerminalExpression.interpret(context: context);
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

